bootloader/ubootenv/env.go

```diff
@@ -101,16 +115,44 @@ func OpenWithFlags(fname string, flags OpenFlags) (*Env, error) {
 		return nil, err
 	}
 
-	if len(contentWithHeader) < headerSize {
-		return nil, fmt.Errorf("cannot open %q: smaller than expected header", fname)
+	// Most systems have SYS_REDUNDAND_ENVIRONMENT=y, so try that first
+	env, err := readEnv(contentWithHeader, flags, true)
```

Comment:
(nitpick) I would write this as:
```
        headerFlagByte := true
	env, err := readEnv(contentWithHeader, flags, true)
	...
	headerFlagByte = false
        env, err := readEnv(contentWithHeader, flags, true)
```
i.e. make it more explicit what this boolean means but maybe I'm overthinking it :)

Reply:
It makes it a bit more readable so why not. Adjusted (to what I think you were going for there)